### ISSUE TITLE
[iOS] How to output Chinese file name?

### ISSUE THREAD

**Issue by cimonk** [tanersener/mobile-ffmpeg#360]
I use this command to do merge video file:
-y -f concat -safe -1 -i "/Users/XXX/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/F07EE09F-A82D-4F3C-AC79-819005BDBD44/tmp/【K萌君】教你做Windows+Ubuntu双系统.txt" -c copy -bsf:a aac_adtstoasc /Users/XXX/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/F07EE09F-A82D-4F3C-AC79-819005BDBD44/Documents/Downloads/20200306_141905981.mp4

But ffmpeg output the error log:
Command execution failed with rc=1 and output=ffmpeg version git-2020-01-25-fd11dd500 Copyright (c) 2000-2020 the FFmpeg developers
  built with Apple clang version 11.0.0 (clang-1100.0.33.17)
  configuration: --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk --prefix=/Users/taner/Projects/mobile-ffmpeg/prebuilt/ios-x86_64-apple-darwin/ffmpeg --enable-version3 --arch=x86_64 --cpu=x86_64 --target-os=darwin --ar=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar --cc=clang --cxx=clang++ --as='clang -arch x86_64 -target x86_64-ios-darwin -march=x86-64 -msse4.2 -mpopcnt -m64 -mtune=intel -DMOBILE_FFMPEG_X86_64 -Wno-unused-function -Wno-deprecated-declarations -fstrict-aliasing -DIOS -DMOBILE_FFMPEG_BUILD_DATE=20200125 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk -O2 -mios-simulator-version-min=12.1 -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator13.2.sdk/usr/include' --ranlib=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib --strip=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip --disable-neon --enable-cross-compile --enable-pic --enable-asm --enable-inline-asm --enable-optimizations --enable-swscale --enable-static --disable-shared --enable-small --disable-v4l2-m2m --disable-outdev=v4l2 --disable-outdev=fbdev --disable-indev=v4l2 --disable-indev=fbdev --disable-openssl --disable-xmm-clobber-test --disable-debug --disable-neon-clobber-test --disable-programs --disable-postproc --disable-doc --disable-htmlpages --disable-manpages --disable-podpages --disable-txtpages --disable-sndio --disable-schannel --disable-securetransport --disable-xlib --disable-cuda --disable-cuvid --disable-nvenc --disable-vaapi --disable-vdpau --disable-appkit --disable-alsa --disable-cuda --disable-cuvid --disable-nvenc --disable-vaapi --disable-vdpau --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-gmp --enable-gnutls --enable-libmp3lame --enable-libass --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libxml2 --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libshine --enable-libspeex --enable-libwavpack --enable-libkvazaar --enable-libilbc --enable-libopus --enable-libsnappy --enable-libsoxr --enable-libaom --enable-libtwolame --disable-sdl2 --enable-zlib --enable-audiotoolbox --enable-coreimage --enable-bzlib --enable-videotoolbox --enable-avfoundation --enable-iconv
  libavutil      56. 38.100 / 56. 38.100
  libavcodec     58. 65.102 / 58. 65.102
  libavformat    58. 35.101 / 58. 35.101
  libavdevice    58.  9.103 / 58.  9.103
  libavfilter     7. 70.101 /  7. 70.101
  libswscale      5.  6.100 /  5.  6.100
  libswresample   3.  6.100 /  3.  6.100
/Users/XXX/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/F07EE09F-A82D-4F3C-AC79-819005BDBD44/tmp/KY`ZWindows+Ubuntu√ÉÀöÔ¨Ç.txt: No such file or directory

The log seems it could not set UTF-8 or unicode argument.

Please help, thanks.

**Comment by chaoskyme** [comment]
Me too. I have tried again and again.

**Comment by tanersener** [comment]
How did you find `"/Users/XXX/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/F07EE09F-A82D-4F3C-AC79-819005BDBD44/tmp/【K萌君】教你做Windows+Ubuntu双系统.txt"` path? 

What happens if you add `【K萌君】教你做Windows+Ubuntu双系统.txt` as a resource in your application try to use it as a resource?

**Comment by chaoskyme** [comment]
@tanersener I run code with Swift:
```swift
MobileFFmpeg.execute("-i \'/var/mobile/Containers/Data/Application/0CE7F4A0-4A4C-45D8-914F-55A01DA9EA9F/Documents/Audios/百花香 - 魏新雨.aac\' -i \'/var/mobile/Containers/Data/Application/0CE7F4A0-4A4C-45D8-914F-55A01DA9EA9F/Documents/Audios/一个人挺好 - 孟颖.aac\' -filter_complex \'[0:0] [1:0]  concat=n=2:v=0:a=1 [a]\' -map [a] /private/var/mobile/Containers/Data/Application/0CE7F4A0-4A4C-45D8-914F-55A01DA9EA9F/tmp/022025FB-278F-4861-8531-24A693D4CCCB.aac")
```
I get an errror like this in debug area.

`ERROR: /var/mobile/Containers/Data/Application/0CE7F4A0-4A4C-45D8-914F-55A01DA9EA9F/Documents/Audios/~¬±√¥ - O‚àû√ã.aac: No such file or directory`

**Comment by tanersener** [comment]
Your issues are not about the library, they are about accessing files with Chinese characters in iOS/tvOS. 

I don't have experience with Chinese characters. So I don't have an idea about the fix. Did you try searching for solutions on `stackoverflow`?

I think you need to try to access those files using a source block like the following one. Did you try that?

```
NSString* path = [[NSBundle mainBundle] pathForResource:@"filename" ofType:@"extension"];
NSLog(@"path is %@",path);
```

**Comment by cimonk** [comment]
> How did you find `"/Users/XXX/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/F07EE09F-A82D-4F3C-AC79-819005BDBD44/tmp/【K萌君】教你做Windows+Ubuntu双系统.txt"` path?
> 
> What happens if you add `【K萌君】教你做Windows+Ubuntu双系统.txt` as a resource in your application try to use it as a resource?

Hi tanersener,

The contents of `【K萌君】教你做Windows+Ubuntu双系统.txt` is
`file '/Users/ike/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/30D65387-BE20-445A-9180-2EDDB7CDBA73/tmp/【K萌君】教你做Windows+Ubuntu双系统[0].flv'
file '/Users/ike/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/30D65387-BE20-445A-9180-2EDDB7CDBA73/tmp/【K萌君】教你做Windows+Ubuntu双系统[1].flv'
file '/Users/ike/Library/Developer/CoreSimulator/Devices/9B0CC62F-D424-42D2-93E9-7D9FF672B29E/data/Containers/Data/Application/30D65387-BE20-445A-9180-2EDDB7CDBA73/tmp/【K萌君】教你做Windows+Ubuntu双系统[2].flv'`

How about iOS Swift use MobileFFmpeg.execute() with Chinese characters?

BTW, my android colleague can use their MobileFFmpeg library to execute with Chinese characters successfully.

**Comment by cimonk** [comment]
> Your issues are not about the library, they are about accessing files with Chinese characters in iOS/tvOS.
> 
> I don't have experience with Chinese characters. So I don't have an idea about the fix. Did you try searching for solutions on `stackoverflow`?
> 
> I think you need to try to access those files using a source block like the following one. Did you try that?
> 
> ```
> NSString* path = [[NSBundle mainBundle] pathForResource:@"filename" ofType:@"extension"];
> NSLog(@"path is %@",path);
> ```

I searched that iOS and FFmpeg with Chinese characters, but I didn't find about it.

I also tried it as below, but it does not work too.
```
NSString* path = [[NSBundle mainBundle] pathForResource:@"filename" ofType:@"extension"];
NSLog(@"path is %@",path);
```

**Comment by chaoskyme** [comment]
I found this problem in function `+ (NSArray*)parseArguments: (NSString*)command `, if use unicode char take from Chinese characters, it will take error in this line code `char currentChar = [command characterAtIndex:i];`.

**Comment by tanersener** [comment]
@chaoskyme Thanks for the analysis. Apparently, there was a bug about parsing unicode parameters in `MobileFFmpeg.parseArguments` function. I fixed that on `development` branch. It will be included in the next release.

Please use `MobileFFmpeg.executeWithArguments` method until this fix released.

**Comment by cimonk** [comment]
Yes!
`MobileFFmpeg.executeWithArguments` is work!
Thank you @tanersener @chaoskyme  !!

**Comment by tanersener** [comment]
Released with `v4.3.2`
